src/system/drivers/profile_change_driver/profile_change_driver.rs

```diff
@@ -0,0 +1,7 @@
+use crate::prelude::*;
+
+#[uniffi::export(with_foreign)]
+#[async_trait::async_trait]
+pub trait ProfileChangeDriver: Send + Sync + std::fmt::Debug {
+    async fn handle_profile_change(&self, changed_profile: Profile);
```

Comment:
this is quite costly, sending the whole Profile. But it allows for simple implementation in hosts.

We should just be mindful that this might result in poor performance. 

An alternative would be, 6 drivers instead of one:
`FactorSourcesChangedDriver`
`AppPreferencesChangedDriver`
`GatewaysChangedDriver`
`AccountsChangedDriver`
`PersonasChangedDriver`
`AuthorizedDappsChangedDriver`

Which would be **drastically** smaller payloads sent. But perhaps not worth the complexity, especially since this is  all in flux and the Profile "API surface" will get smaller over time once more and more logic into Sargon.

Also, Profile payload being big is probably just a problem for Profiles with 10+ accounts which most user do not have.

Reply:
I don't think this applies here, the profile change is async. So what would happen is that the async operation of modifying a profile will take slightly longer, it should not impact the overall performance while using the app.

And as mentioned, **this is the first step**, going straight away to the final solution is too costly from the development perspective.